src/main/java/org/folio/qm/validation/LeaderValidationRule.java

```diff
@@ -0,0 +1,73 @@
+package org.folio.qm.validation;
+
+
+import static org.folio.qm.converter.elements.LeaderItem.BASE_ADDRESS;
+import static org.folio.qm.converter.elements.LeaderItem.CODING_SCHEME;
+import static org.folio.qm.converter.elements.LeaderItem.ENTRY_MAP_20;
+import static org.folio.qm.converter.elements.LeaderItem.ENTRY_MAP_21;
+import static org.folio.qm.converter.elements.LeaderItem.ENTRY_MAP_22;
+import static org.folio.qm.converter.elements.LeaderItem.ENTRY_MAP_23;
+import static org.folio.qm.converter.elements.LeaderItem.INDICATOR_COUNT;
+import static org.folio.qm.converter.elements.LeaderItem.RECORD_LENGTH;
+import static org.folio.qm.converter.elements.LeaderItem.SUBFIELD_CODE_LENGTH;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.folio.qm.converter.elements.Constants;
+import org.folio.qm.converter.elements.LeaderItem;
+import org.folio.qm.domain.dto.MarcFormat;
+
+public interface LeaderValidationRule {
+
+  boolean supportFormat(MarcFormat marcFormat);
+
+  Optional<ValidationError> validate(String leader);
+
+  default Optional<ValidationError> commonLeaderValidation(String leader) {
+    Optional<ValidationError> error = validateLeaderLength(leader);
+    if (error.isPresent()) {
+      return error;
+    }
+    error = validateLeaderNumberFields(leader, RECORD_LENGTH.getPosition(), RECORD_LENGTH.getLength());
+    if (error.isPresent()) {
+      return error;
+    }
+    error = validateLeaderNumberFields(leader, BASE_ADDRESS.getPosition(), BASE_ADDRESS.getLength());
+    if (error.isPresent()) {
+      return error;
+    }
+    return validateLeaderFieldsRestrictions(leader, List.of(CODING_SCHEME, INDICATOR_COUNT, SUBFIELD_CODE_LENGTH,
+      ENTRY_MAP_20, ENTRY_MAP_21, ENTRY_MAP_22, ENTRY_MAP_23));
+  }
```

Comment:
could you change it to 
return  Stream.of(
        validateLeaderLength(leader),
        validateLeaderNumberFields(leader, RECORD_LENGTH.getPosition(), RECORD_LENGTH.getLength()),
        validateLeaderNumberFields(leader, BASE_ADDRESS.getPosition(), BASE_ADDRESS.getLength()),
        validateLeaderFieldsRestrictions(leader, List.of(CODING_SCHEME, INDICATOR_COUNT, SUBFIELD_CODE_LENGTH,
            ENTRY_MAP_20, ENTRY_MAP_21, ENTRY_MAP_22, ENTRY_MAP_23)))
        .filter(Optional::isPresent)
        .map(Optional::get)
        .findFirst();